qiskit/circuit/parameterexpression.py

```diff
@@ -375,7 +375,20 @@ def __deepcopy__(self, memo=None):
         return self
 
     def __eq__(self, other):
+        """Check if this parameter expression is equal to another parameter expression
+           or a fixed value (only if this is a bound expression).
+        Args:
+            other (ParameterExpression or a number):
+                Parameter expression or numeric constant used for comparison
+        Returns:
+            bool: result of the comparison
+        """
         from sympy import srepr
-        return (isinstance(other, ParameterExpression)
-                and self.parameters == other.parameters
-                and srepr(self._symbol_expr) == srepr(other._symbol_expr))
+        if isinstance(other, ParameterExpression):
+            return (self.parameters == other.parameters
+                    and srepr(self._symbol_expr) == srepr(other._symbol_expr))
+        elif isinstance(other, numbers.Number):
+            return (len(self.parameters) == 0
+                    and complex(self._symbol_expr) == other)
+        else:
+            return False
```

Comment:
we can skip the `else` block since this is only reached if none of the previous `if` blocks are being entered 
```suggestion
        return False
```